src/cli.zig

```diff
@@ -185,6 +185,7 @@ pub const Arguments = struct {
         clap.parseParam("--jsx-fragment <STR>              Changes the function called when compiling JSX fragments") catch unreachable,
         clap.parseParam("--jsx-import-source <STR>         Declares the module specifier to be used for importing the jsx and jsxs factory functions. Default: \"react\"") catch unreachable,
         clap.parseParam("--jsx-runtime <STR>               \"automatic\" (default) or \"classic\"") catch unreachable,
+        clap.parseParam("--ignore-annotations             Ignore tree-shaking annotations such as @__PURE__") catch unreachable,
```

Comment:
> --ignore-annotation

To someone who doesn't know much about bundlers, this  CLI flag isn't clear.  What is an annotation? A TypeScript type? A decorator? A JSDoc comment? A license comment?

Why not `--ignore-dce-annotation`? Or maybe `--ignore-pure-comments`

Reply:
Bikeshed the new name, should still alias to support what esbuild does: https://esbuild.github.io/api/#ignore-annotations

Reply:
I think it is clear if you read the description of the option: `Ignore tree-shaking annotations such as @__PURE__`

The key words being `tree-shaking` and `@__PURE__`. If that isn't enough, a single google search will take you to a page explaining it.

![image](https://github.com/user-attachments/assets/d5389990-5576-4d15-8522-a7bcc6dd7dfb)

I did not add anything to our docs in this PR explaining these flags.

I would be extremely surprised if someone reading a line talking about `tree-shaking annotations` to misinterpret it as anything related to type definitions (bun already ignores/does not check types), jsdoc (comments are also ignored), and decorators (not sure what ignoring decorators would even do).

> Why not --ignore-dce-annotation?

The flag was chosen to be the same as esbuild. I would be okay with changing it to `--ignore-dce-annotations`.

> `--ignore-pure-comments`

Do not recommend this naming since `@__PURE__` is not the only form of annotations. One day I want to support `@__NO_SIDE_EFFECTS__`. "dce annotations" (or simply annotations) is a more correct term.